### ISSUE TITLE
Neo4j and cypher-shell bug.

### ISSUE THREAD

**Issue by SundbergBrian** [neo4j/neo4j#11771]
With no change to development environment cypher-shell suddenly no longer functioned. Cypher-shell, when running a cql file that has been reliable in the past, returns this error :

> sudo neo4j-shell -v -path /var/lib/neo4j/data/databases/graph.db --file /home/dart/Desktop/crawl/your-file.cql
> ERROR (-v for expanded information):
> 	Error starting org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory, /var/lib/neo4j/data/databases/graph.db
> java.lang.RuntimeException: Error starting org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory, /var/lib/neo4j/data/databases/graph.db
> 	at org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:211)
> 	at org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory.newFacade(GraphDatabaseFacadeFactory.java:126)
> 	at org.neo4j.graphdb.factory.GraphDatabaseFactory.newDatabase(GraphDatabaseFactory.java:137)
> 	at org.neo4j.graphdb.factory.GraphDatabaseFactory.newEmbeddedDatabase(GraphDatabaseFactory.java:130)
> 	at org.neo4j.graphdb.factory.GraphDatabaseFactory$1.newDatabase(GraphDatabaseFactory.java:107)
> 	at org.neo4j.graphdb.factory.GraphDatabaseBuilder.newGraphDatabase(GraphDatabaseBuilder.java:193)
> 	at org.neo4j.shell.kernel.GraphDatabaseShellServer.instantiateGraphDb(GraphDatabaseShellServer.java:228)
> 	at org.neo4j.shell.kernel.GraphDatabaseShellServer.<init>(GraphDatabaseShellServer.java:75)
> 	at org.neo4j.shell.StartClient.getGraphDatabaseShellServer(StartClient.java:318)
> 	at org.neo4j.shell.StartClient.tryStartLocalServerAndClient(StartClient.java:294)
> 	at org.neo4j.shell.StartClient.startLocal(StartClient.java:282)
> 	at org.neo4j.shell.StartClient.start(StartClient.java:213)
> 	at org.neo4j.shell.StartClient.main(StartClient.java:147)
> Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter@3e08ff24' was successfully initialized, but failed to start. Please see the attached cause exception "Store and its lock file has been locked by another process: /var/lib/neo4j/data/databases/graph.db/store_lock. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)".
> 	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:466)
> 	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:107)
> 	at org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:207)
> 	... 12 more
> Caused by: org.neo4j.kernel.StoreLockException: Store and its lock file has been locked by another process: /var/lib/neo4j/data/databases/graph.db/store_lock. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)
> 	at org.neo4j.kernel.internal.locker.StoreLocker.storeLockException(StoreLocker.java:116)
> 	at org.neo4j.kernel.internal.locker.StoreLocker.checkLock(StoreLocker.java:92)
> 	at org.neo4j.kernel.internal.locker.GlobalStoreLocker.checkLock(GlobalStoreLocker.java:60)
> 	at org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter.start(StoreLockerLifecycleAdapter.java:36)
> 	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:445)
> 	... 14 more
> 
>  -host      Domain name or IP of host to connect to (default: localhost)
>  -port      Port of host to connect to (default: 1337)
>  -name      RMI name, i.e. rmi://<host>:<port>/<name> (default: shell)
>  -pid       Process ID to connect to
>  -c         Command line to execute. After executing it the shell exits
>  -file      File containing commands to execute, or '-' to read from stdin. After executing it the shell exits
>  -readonly  Connect in readonly mode (only for connecting with -path)
>  -path      Points to a neo4j db path so that a local server can be started there
>  -config    Points to a config file when starting a local server
> 
> Example arguments for remote:
> 	-port 1337
> 	-host 192.168.1.234 -port 1337 -name shell
> 	-host localhost -readonly
> 	...or no arguments for default values
> Example arguments for local:
> 	-path /path/to/db
> 	-path /path/to/db -config /path/to/neo4j.config
> 	-path /path/to/db -readonly

### ( Note 1 )This is with a fresh installation of neo4j after I used synaptic package manager to remove all neo4j related packages and reinstalled them, fresh and unadulterated.

- Neo4j version: 3.3.5
- Operating system: Ubuntu 17.04
- API/Driver: Cypher

### Steps to reproduce
1. Remove neo4j from system using package manager.
2. Install a fresh version of neo4j.
3. Reboot
4. Run Cypher-shell command once, successfully.
5. Run same Cypher-shell again, fail.

### Expected behavior
Error does not present itself, and the commands included in the cql file get executed by the neo4j server.

### Actual behavior
Error present itself, commands to not reach server.

### (Note 2)
After running 

> sudo service neo4j stop

then

> sudo neo4j stop

and confirming both were not running, I was was still able to acess to "localhost:7474" web interface and still create and query nodes, that are persistant. Neo4j's installation package must be installing multiple instances of the server, or be magic.

**Comment by Fohlen** [comment]
I am encountering this as well. weird.

**Comment by craigtaverner** [comment]
Can you clarify something for me? In the title of the issue and in the description the term `cypher-shell` is used. However in the actual command listed above you instead run `neo4j-shell`. That is a very different application, one that has been deprecated since Neo4j 3.0 in favour of `cypher-shell`. Before we can act on this issue it would be good to know which shell you really meant to run. If it is `neo4j-shell` can you perhaps edit the title for clarity.

**Comment by Fohlen** [comment]
@craigtaverner for me it is the `cypher-shell`. I have found out what makes this error occur: if your Neo4j instance currently has load on it (say pending transactions) then `cypher-shell` will try to connect but somehow fail, exit, and prompt you with your actual shell again. As soon as all transactions are finished entering `cypher-shell` does work like a charm again.

**Comment by craigtaverner** [comment]
Hi @Fohlen. Your description sounds like a different bug than the one reported above, with a different command and different symptoms. Is it possible for you to open a separate issue for what you have observed? Bugs in cypher-shell or in cypher-shell connectivity are worked on by different teams than the bug here which appears to be an embedded database issue with offline use.

**Comment by Fohlen** [comment]
Hey @craigtaverner where would you like me to report this? This should also be quite easy to reproduce with the `neo4j` docker release!

**Comment by tinwelint** [comment]
For the original issue, using `neo4j-shell -path ...` would not even try to connect to a running instance but always try to start the database itself and if it was already running you would get that exception about the store lock. Just make sure that the database isn't running and you should be fine. The neo4j-shell is no longer present in all currently supported versions, either.
